pkg/oci/client/load_balancer.go

```diff
@@ -0,0 +1,301 @@
+	CreateListener(ctx context.Context, lbID, name string, details loadbalancer.ListenerDetails) (string, error)
+	DeleteListener(ctx context.Context, lbID, name string) (string, error)
+
+	AwaitWorkRequest(ctx context.Context, id string) (*loadbalancer.WorkRequest, error)
+
+	// TODO:
+	// DeleteListener
+	// UpdateListener
+	// GetSubnetsForNodes
+}
+
+func (c *client) GetLoadBalancer(ctx context.Context, id string) (*loadbalancer.LoadBalancer, error) {
+	resp, err := c.loadbalancer.GetLoadBalancer(ctx, loadbalancer.GetLoadBalancerRequest{
+		LoadBalancerId: &id,
+	})
+	incRequestCounter(err, getVerb, loadBalancerResource)
+
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return &resp.LoadBalancer, nil
+}
+
+func (c *client) GetLoadBalancerByName(ctx context.Context, name string) (*loadbalancer.LoadBalancer, error) {
+	var page *string
+	for {
+		resp, err := c.loadbalancer.ListLoadBalancers(ctx, loadbalancer.ListLoadBalancersRequest{
+			CompartmentId: &c.config.Auth.CompartmentOCID,
+			DisplayName:   &name,
+			Page:          page,
+		})
+		incRequestCounter(err, listVerb, loadBalancerResource)
+
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		for _, lb := range resp.Items {
+			if *lb.DisplayName == name {
+				return &lb, nil
+			}
+		}
+		if page = resp.OpcNextPage; page == nil {
+			break
+		}
+	}
+
+	return nil, errors.WithStack(errNotFound)
+}
+
+func (c *client) CreateLoadBalancer(ctx context.Context, details loadbalancer.CreateLoadBalancerDetails) (string, error) {
+	resp, err := c.loadbalancer.CreateLoadBalancer(ctx, loadbalancer.CreateLoadBalancerRequest{
+		CreateLoadBalancerDetails: details,
+	})
+	incRequestCounter(err, createVerb, loadBalancerResource)
+
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	return *resp.OpcWorkRequestId, nil
+}
+
+func (c *client) DeleteLoadBalancer(ctx context.Context, id string) (string, error) {
+	resp, err := c.loadbalancer.DeleteLoadBalancer(ctx, loadbalancer.DeleteLoadBalancerRequest{
+		LoadBalancerId: &id,
+	})
+	incRequestCounter(err, deleteVerb, loadBalancerResource)
+
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	return *resp.OpcWorkRequestId, nil
+}
+
+func (c *client) GetCertificateByName(ctx context.Context, lbID, name string) (*loadbalancer.Certificate, error) {
+	resp, err := c.loadbalancer.ListCertificates(ctx, loadbalancer.ListCertificatesRequest{
+		LoadBalancerId: &lbID,
+	})
+	incRequestCounter(err, listVerb, certificateResource)
+
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	for _, cert := range resp.Items {
+		if *cert.CertificateName == name {
+			return &cert, nil
+		}
+	}
+	return nil, errors.WithStack(errNotFound)
+}
+
+func (c *client) CreateCertificate(ctx context.Context, lbID, certificate, key string) (string, error) {
+	// TODO(apryde): We currently don't have a mechanism for supplying
```

Comment:
Is there a ticket tracking this?

Reply:
https://github.com/oracle/oci-cloud-controller-manager/issues/149